src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -28,73 +29,69 @@
 
 class MissingOrgMemberSerializer(Serializer):
     def serialize(self, obj, attrs, user, **kwargs):
-        return {"email": obj.email, "externalId": obj.external_id, "commitCount": obj.commit_count}
+        return {"email": obj.email, "externalId": obj.external_id, "commitCount": obj.commit__count}
 
 
 class MissingMembersPermission(OrganizationPermission):
     scope_map = {"GET": ["org:write"]}
 
 
-@region_silo_endpoint
-class OrganizationMissingMembersEndpoint(OrganizationEndpoint):
-    publish_status = {
-        "GET": ApiPublishStatus.UNKNOWN,
-    }
-    permission_classes = (MissingMembersPermission,)
-
-    def _get_missing_members(
-        self, organization: Organization, provider: str, integration_ids: Sequence[int]
-    ) -> QuerySet[CommitAuthor]:
-        member_emails = set(
-            organization.member_set.exclude(email=None).values_list("email", flat=True)
-        )
-        member_emails.update(
-            set(
-                organization.member_set.exclude(user_email=None).values_list(
-                    "user_email", flat=True
-                )
-            )
-        )
-        nonmember_authors = CommitAuthor.objects.filter(organization_id=organization.id).exclude(
-            Q(email__in=member_emails) | Q(external_id=None)
+def _get_missing_organization_members(
+    organization: Organization, provider: str, integration_ids: Sequence[int]
+) -> QuerySet[WithAnnotations[CommitAuthor]]:
+    member_emails = set(organization.member_set.exclude(email=None).values_list("email", flat=True))
+    member_emails.update(
+        set(organization.member_set.exclude(user_email=None).values_list("user_email", flat=True))
+    )
+    nonmember_authors = CommitAuthor.objects.filter(organization_id=organization.id).exclude(
+        Q(email__in=member_emails) | Q(external_id=None)
+    )
+
+    org_repos = Repository.objects.filter(
+        provider="integrations:" + provider,
+        organization_id=organization.id,
+        integration_id__in=integration_ids,
+    ).values_list("id", flat=True)
+
+    return (
+        nonmember_authors.filter(
+            commit__repository_id__in=set(org_repos),
```

Comment:
Oh interesting.

Not sure this is really an issue, but I wonder if there's a way to make this more efficient - rather than querying for both tables, and instead FK-ing the tables or something 🤔

Reply:
Good point. They're relatively small queries and since this is offline computation I think we can get away with the current logic.